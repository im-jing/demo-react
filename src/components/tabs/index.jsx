import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './index.less';

export class Tabs extends Component {
  constructor(props, context) {
    super(props, context);

    this.state = {
      activeIdx: props.activeTabIdx,
    };
  }

  handleTabClick = (idx) => {
    this.setState({
      activeIdx: idx,
    });
  }

  renderChildrenTabs = () => {
    const { children } = this.props;
    const { activeIdx } = this.state;

    return React.Children.map(children, (child, idx) => {
      const tabClass = activeIdx === idx ? 'tab-item active' : 'tab-item';

      return (
        <li>
          <button
            type="button"
            className={tabClass}
            onClick={() => this.handleTabClick(idx)}
          >
            {child.props.name}
          </button>
        </li>
      );
    });
  }

  renderTabActiveContent = () => {
    const { children } = this.props;
    const { activeIdx } = this.state;

    if (children[activeIdx]) {
      return children[activeIdx].props.children;
    }
  }

  render() {
    return (
      <React.Fragment>
        <ul className="tabs">
          {this.renderChildrenTabs()}
        </ul>

        <div className="tabs-content">
          {this.renderTabActiveContent()}
        </div>
      </React.Fragment>
    );
  }
}
Tabs.propTypes = {
  activeTabIdx: PropTypes.number.isRequired,
  children: PropTypes.node.isRequired,
};

export const Tab = ({ children }) => <React.Fragment>{ children }</React.Fragment>;
Tab.propTypes = {
  children: PropTypes.node.isRequired,
};
