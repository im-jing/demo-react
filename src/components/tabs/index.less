.tabs {
  margin: 0;
  padding: 0;
}
.tabs li {
  display: inline-block;
}
.tab-item {
  display: inline-block;
  padding: 5px 15px;
}
.tab-item.active {
  background-color: cornflowerblue;
}
.tab-item-content {
  display: none;
}
.tab-item-content.active {
  display: inline-block;
}