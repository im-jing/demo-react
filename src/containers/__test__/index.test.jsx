import Home from '../index';
import { Tabs, Tab } from '../../components/tabs/index';

jest.mock('../../request/api');

const wrapper = shallow(<Home />);

describe('Home Page', () => {
  it('render the Home to match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the tabs component', () => {
    expect(wrapper.children(Tabs).length).toEqual(1);
    expect(wrapper.find(Tab).length).toEqual(3);
  });

  // it('fetch locations api and render the result on mount', (done) => {
  //   const wrapper = shallow(<Home />);

  //   // expect(wrapper.find())
  // });
});
