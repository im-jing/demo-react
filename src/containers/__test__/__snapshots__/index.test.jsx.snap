// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page render the Home to match the snapshot 1`] = `
<Fragment>
  <Loading />
</Fragment>
`;
